### DISCUSSION TITLE
[꿀팁] Normalization vs Regularization 차이

### DISCUSSION THREAD

**Discussion by areum514** [bcaitech1/p4-opt-5-vibrhanium-#26] [꿀팁]
둘 다 저희가 흔히 아는 `정규화` 라고 부르는 애들인데, 명확히 따지자면 다른 개념입니다.

각각 정규화를 하기 위해 다루는 것들이 다른데, 간략하게 보자면 아래와 같습니다.

- Normalization : Data에 대한 조정
- Regularization : Model에 대한 정규화 (규제)

### Normalization에 관하여

- `Feature Scaling`

 흔히 저희가 아는 Normalize는 다르게 표현하자면 Feature Scaling 이라고 볼 수 있습니다. **데이터가 가진 Feature끼리의 값이 지나치게 큰 차이가 나타나는 등의 경우, data를 조정해주어 비교가 수월하도록 하는 것**들을 일컫습니다. 만약 normalize되지 않은 data, 즉 noise를 처리해주지 않은 데이터를 사용하여 모델에게 학습하라고 집어넣을 경우, 모델을 과적합시키는 원인 중 하나가 될 수 있습니다.

> **왜 문제가 될까?**
>
> ---
>
> 예를들어 건물가격을 예측할 때 건물의 노후연식과 방의 개수를 가지고 분석한다 하면, 방의 개수는 0~4 과 같이 적은 값을 갖지만, 그에 반해 연식은 0~40 혹은 그 이상의 값을 갖게 되어 Feature끼리의 scale 차이가 크게 발생하게 됩니다.
>
> ​	x : 0 ~ 40 (years)
>
> ​	y : 0 ~ 4 (rooms)
>
> 이런 데이터를 Normalize하지 않고 그대로 사용할 경우 data point끼리 더욱 가깝게 존재하게 되고 제대로된 비교를 하지 못할 수 있어 overfitting의 이유 중 하나가 됩니다.

 조금만 자세히 알아보자면 Feature Scaling은 `Normalize`와 `Standardize`로 나뉩니다.

- Normalize  (min-max scaling)

   보통은 0~1 사이의 값으로 범주를 일치시키는 작업을 말합니다.
  가장 대표적인 방법으로는 `minmax scaling `이 있습니다. 데이터끼리 비교를 쉽게 만들어 처리가 간단하다는 장점이 있습니다.

   주의할 점으로는 min-max 범위 밖에 새로운 데이터가 input되는 경우 out-of-bound가 발생할 수 있습니다.
  $$
  x =  \frac {x - x_{min}} {x_{max} - x_{min}}
  $$

- Standardize (z-transformation)

   고등학교 때 배운 표준정규분포를 구하는 식으로 구하면 됩니다. z-score라는 점수를 사용하며 평균과 표준편차를 통해 구할 수 있습니다.

   보통 최대값과 최소값을 모를 때 유용하게 사용할 수 있고, 표준편차 대신 절대평균편차를 사용하는 경우 outlier에 대해 보다 잘(엄격히) 처리하는 경향이 있습니다.
  $$
  x = \frac {x-m} {\varrho}  (m: 평균, \varrho: 표준편차)
  $$

  > **절대평균편차란?**
  >
  > ---
  >
  > 각 측정값과 평균 사이의 거리의 평균으로, 자료 집합의 산포도를 나타내는 지표
  >
  > 1. 평균을 구한다.
  > 2. 각 측정값이 평균으로부터 얼마나 떨어져있는지를 구한 후 절대값을 씌운다.
  >    (이 값을 절대편차 라고 부릅니다.)
  > 3. 모든 편차들을 더한다.
  > 4. 편차의 합을 측정값의 개수로 나눈다.

### Regularization

- `Generalization`

 모델의 학습에 있어 학습데이터에만 치중하지 않고 조금 더 일반적으로 판단하도록 하는 방법들을 말합니다. 다른 말로 표현하면 Overfitting을 막고 Generalize된 모델을 통해 일반화 성능을 높이는 방법이라고 할 수 있습니다.

가장 대표적인 방법들을 몇 가지 나열해보면 아래와 같습니다.

#### 	Weight Decay 관점

 	가중치가 클수록 overfitting을 일으킬 요인이 크기때문에 패널티를 부여해야 한다는 관점에서 출발합니다.

> Weight에 가중치(패널티)를 부여함으로써 모델학습을 규제

$$
E(W) = MSE_{train} + \frac 1 2 * \lambda W^TW
$$

 데이터가 단순하고 모델이 복잡하면, 학습을 하면서 굉장히 작은 값이었던 weight들이 점점 증가하게 되면서 overfitting문제가 발생하게 됩니다. 즉, 학습데이터의 영향을 많이 받게 되고 모델이 그에 더 잘 들어맞게 바뀌게 되는 성향을 띄게 되고, 이 것을 'local noise에 영향을 크게 받아 outlier에 모델이 맞춰지는 현상' 이라고 합니다.

> gradient descent 일반식

$$
w_{i+1} = w_{i} - l * \nabla_w{E(w_i)}
$$

1. L1 Regularization (Lasso regression) - 입력 차원을 낮춰줌(Feature Selection에 유용)

2. L2 Regularization (Ridge regression) - 보통 더 잘 학습시켜줌(Model Training에 유용)

   ---

   #### 그 외

3. Dropout - node 탈락시키기 (성능은 좋지만 학습 속도가 느려진다는 단점이 있음)

     다른 unit이 잘못 학습한 것을 다른 unit이 보완하도록 학습하는 경향을 끊어줌으로써 성능을 향상시키는 방법, 마찬가지로 overfitting을 방지하며 Regularization 효과가 있다.

4. Batch Normalization (Dropout과 비슷한 효과, Dropout 대신 BN으로 자주 사용한다.)

    미니배치 단위에서 정규화를 수행하는 방법으로, 각각의 Feature들을 독립적으로 정규화하는 방식으로 진행합니다. (각각의 Feature들의 mean, variance를 0과 1로 정규화하는 것)

5. Early Stopping - 학습을 반복해도 loss가 안줄면 (validation loss가 증가하면) 학습을 종료하는 방법.

6. Data Augmentation - 데이터를 부풀리기(왜곡, 반전, 회전 등등)

출처 ustage 2조 이종혁 t1162

**Comment by JaegyeongByun** [comment]
꿀팁 감사합니다!!
